src/xinteractiveshell.hpp

```diff
@@ -0,0 +1,73 @@
+#include "pybind11/pybind11.h"
+#include "xdisplay.hpp"
+
+#ifdef __GNUC__
+    #pragma GCC diagnostic ignored "-Wattributes"
+#endif
+
+namespace py = pybind11;
+using namespace pybind11::literals;
+
+namespace xpyt 
+{
+
+    struct hooks_object
+    {
+        hooks_object() {}
+        static inline void show_in_pager(py::str data, py::kwargs)
+        {
+            xdisplay(py::dict("text/plain"_a=data), {}, {}, py::dict(), py::none(), py::none(), false, true);
+        }
+
+    };
+
+    class XInteractiveShell {
+
+    private:
+        py::module _ipy_process;
+        py::module _magic_core;
+        py::module _magics_module;
+        py::module _extension_module;
```

Comment:
Naming convention: we usually use `m_xxx` for attributes names.

The private section should come after the public one (most of the time you want to read the API, not the implementation details)